packages/SwingSet/src/kernel/kernel.js

```diff
@@ -375,75 +386,164 @@ export default function buildKernel(
    * @returns {Promise<DeliveryStatus>}
    */
   async function deliverAndLogToVat(vatID, kd, vd) {
+    vatRequestedTermination = undefined;
+    illegalSyscall = undefined;
     assert(vatWarehouse.lookup(vatID));
     // Ensure that the vatSlogger is available before clist translation.
     const vs = kernelSlog.provideVatSlogger(vatID).vatSlog;
     try {
       /** @type { VatDeliveryResult } */
       const deliveryResult = await vatWarehouse.deliverToVat(vatID, kd, vd, vs);
-
       insistVatDeliveryResult(deliveryResult);
       // const [ ok, problem, usage ] = deliveryResult;
-      if (deliveryResult[0] === 'ok') {
-        return { metering: deliveryResult[2] };
-      } else {
-        // probably a hard metering fault, or a bug in the vat's dispatch()
-        return { terminate: deliveryResult[1] }; // might be dead
+
+      const status = { illegalSyscall, vatRequestedTermination };
```

Comment:
Nit:
```suggestion
      /** @type {DeliveryStatus} */
      const status = { illegalSyscall, vatRequestedTermination };
```

And remove the `@returns` JSDoc from the function, to let tsc flow the type.

Reply:
The primary purpose of JSDoc annotations is to document the code for a human reader.  The way TypeScript-in-JSDoc piggybacks on top of these annotations is convenient but it's a hack and this is one of the places where the seams show.  Removing the `@returns` for the sake of allowing `tsc` to infer the type is letting the tail wag the dog.  Can we have an `@returns` annotation that omits the `{...}` type info but retains the human-readable text or will `tsc` stupidly complain if we do that?

Reply:
I think it may be able to do what you ask, but I'm not sure. There's always the "solution" to type both the return and this declaration. My concern with having it solely in the returns was that the inferred type for the declaration wasn't correct: it only considered these 2 props, and the other props were completely undocumented even when adding them (and since we run is non-strict type mode, TS happily let you add props that aren't on the declared on the object)